### PR TITLE
Support multiple ecosystems per project

### PR DESCRIPTION
**Pull request by IvanIvanoff**
## Changes

> Note: The ecosystems will be imported by the backend team from a list of manually curated ecosystems.

### Extend allProjectsByFunction

Get all projects with a given ecosystem. If more than 1 ecosystem is provided, it returns the projects that have all the ecosystems.
```graphql
{
  allProjectsByFunction(function: "{\"name\": \"ecosystems\", \"args\": {\"ecosystems\": [\"ethereum\"]}}") {
    projects {
      slug
      name
    }
    stats {
      projectsCount
    }
  }
}
```
```json
{
  "data": {
    "allProjectsByFunction": {
      "projects": [
        {
          "name": "j2dGtJd2gj",
          "slug": "gmU93so/cy"
        }
      ],
      "stats": {
        "projectsCount": 1
      }
    }
  }
}
```
---

### Extend the getMetric selector

Get data separately for all projects with an ecosystem.
```graphql
{
  getMetric(metric: "dev_activity_1d") {
    timeseriesDataPerSlug(selector: {ecosystems: ["ethereum"]}, from: "utc_now-2d", to: "utc_now") {
      data {
        slug
        value
      }
      datetime
    }
  }
}
```
```json
{
  "data": {
    "getMetric": {
      "timeseriesDataPerSlug": [
        {
          "data": [
            {
              "slug": "ethereum",
              "value": 53
            },
            {
              "slug": "santiment",
              "value": 0
            }
          ],
          "datetime": "2024-01-21T00:00:00Z"
        },
        {
          "data": [
            {
              "slug": "ethereum",
              "value": 334
            },
            {
              "slug": "santiment",
              "value": 18
            }
          ],
          "datetime": "2024-01-22T00:00:00Z"
        }
      ]
    }
  }
}
```

Get aggregated data for all projects with an ecosystem
```graphql
{
  getMetric(metric: "dev_activity_1d") {
    timeseriesData(selector: {ecosystems: ["ethereum"]}, from: "utc_now-2d", to: "utc_now") {
      value
      datetime
    }
  }
}
```
```json
{
  "data": {
    "getMetric": {
      "timeseriesData": [
        {
          "datetime": "2024-01-21T00:00:00Z",
          "value": 1358
        },
        {
          "datetime": "2024-01-22T00:00:00Z",
          "value": 9003
        }
      ]
    }
  }
}
```

---
### Add new getEcosystems API
```graphql
{
  getEcosystems {
    name
    projects{
      slug
    }
  }
}
```
```json
{
  "data": {
    "getEcosystems": [
      {
        "name": "bitcoin",
        "projects": [
          {
            "slug": "santiment"
          },
          {
            "slug": "h/vyT4TOxG"
          },
          {
            "slug": "gmU93so/cy"
          },
          {
            "slug": "D18tye+n1w"
          },
          {
            "slug": "AXmYWOYQPR"
          }
        ]
      },
      {
        "name": "ethereum",
        "projects": [
          {
            "slug": "santiment"
          },
          {
            "slug": "gmU93so/cy"
          },
          {
            "slug": "ethereum"
          }
        ]
      }
    ]
  }
}
```

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have tried to find clearer solution before commenting hard-to-understand parts of code
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: lib/sanbase/project/ecosystem/ecosystem.ex [hunk 1]
```diff
@@ -0,0 +1,117 @@
+defmodule Sanbase.Ecosystem do
+  @moduledoc ~s"""
+  Module for managing the "github_organziations" postgres table
```
Comment (tspenov) [inline lib/sanbase/project/ecosystem/ecosystem.ex:3]: copy-pasta

File: lib/sanbase/project/ecosystem/ecosystem.ex [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      on_replace: :delete,
+      on_delete: :delete_all
+    )
+
+    timestamps()
+  end
+
+  @doc false
+  def changeset(%__MODULE__{} = market_segment, attrs \\ %{}) do
+    market_segment
```
Comment (tspenov) [inline lib/sanbase/project/ecosystem/ecosystem.ex:33]: ecosystem?

File: lib/sanbase/user_lists/watchlist_function/watchlist_function.ex [hunk 3]
```diff
@@ -97,6 +97,15 @@ defmodule Sanbase.WatchlistFunction do
     end
   end
 
+  def valid_function?(%__MODULE__{name: "ecosystems", args: args} = fun, opts) do
+    market_segment = Map.get(args, "ecosystems") || Map.fetch!(args, :ecosystems)
```
Comment (tspenov) [inline lib/sanbase/user_lists/watchlist_function/watchlist_function.ex:101]: why it is named `marked_segments` ?

File: lib/sanbase/user_lists/watchlist_function/watchlist_function.ex [hunk 4]
```diff
@@ -211,6 +220,22 @@ defmodule Sanbase.WatchlistFunction do
      }}
   end
 
+  def evaluate(%__MODULE__{name: "ecosystems", args: args}) do
+    ecosystems =
+      Map.get(args, "ecosystems") ||
+        Map.fetch!(args, :ecosystems)
+        |> IO.inspect(label: "ecosystems")
```
Comment (tspenov) [inline lib/sanbase/user_lists/watchlist_function/watchlist_function.ex:227]: inspect left

Comment (IvanIvanoff) [reply lib/sanbase/user_lists/watchlist_function/watchlist_function.ex:227]: That's in a `backup` commit. I usually push multiple times so I don't lose anything and when I rebase, I clean these inspects.

**Pull request merged**